src/app/clusters/media-playback-server/media-playback-server.cpp

```diff
@@ -39,74 +39,158 @@
  ******************************************************************************/
 
 #include <app/Command.h>
+#include <app/clusters/media-playback-server/media-playback-server.h>
+#include <app/common/gen/attribute-id.h>
+#include <app/common/gen/attribute-type.h>
+#include <app/common/gen/cluster-id.h>
+#include <app/common/gen/command-id.h>
+#include <app/common/gen/enums.h>
 #include <app/util/af.h>
 
-bool emberAfMediaPlaybackClusterFastForwardRequestCallback(chip::app::Command * commandObj)
+EmberAfMediaPlaybackStatus mediaPlaybackClusterSendMediaPlaybackRequest(MediaPlaybackRequest mediaPlaybackRequest,
+                                                                        uint64_t deltaPositionMilliseconds);
+
+uint8_t mediaPlaybackClusterPlaybackState;
+
+static void writePlaybackState(chip::EndpointId endpoint, uint8_t playbackState)
+{
+    EmberAfStatus status =
+        emberAfWriteServerAttribute(endpoint, ZCL_MEDIA_PLAYBACK_CLUSTER_ID, ZCL_MEDIA_PLAYBACK_STATE_ATTRIBUTE_ID,
+                                    (uint8_t *) &playbackState, ZCL_INT8U_ATTRIBUTE_TYPE);
+    if (status != EMBER_ZCL_STATUS_SUCCESS)
+    {
+        ChipLogError(Zcl, "Failed to store media playback attribute.");
+    }
+}
+
+static uint8_t readPlaybackStatus(chip::EndpointId endpoint)
 {
-    EmberAfStatus status = EMBER_ZCL_STATUS_SUCCESS;
-    emberAfSendImmediateDefaultResponse(status);
+    uint8_t playbackState;
+    EmberAfStatus status =
+        emberAfReadServerAttribute(endpoint, ZCL_MEDIA_PLAYBACK_CLUSTER_ID, ZCL_MEDIA_PLAYBACK_STATE_ATTRIBUTE_ID,
+                                   (uint8_t *) &playbackState, sizeof(uint8_t));
+    if (status != EMBER_ZCL_STATUS_SUCCESS)
+    {
+        ChipLogError(Zcl, "Failed to read media playback attribute.");
+    }
+
+    return playbackState;
+}
+
+void storeNewPlaybackState(chip::EndpointId endpoint, uint8_t newPlaybackState)
+{
+    mediaPlaybackClusterPlaybackState = readPlaybackStatus(endpoint);
+
+    if (mediaPlaybackClusterPlaybackState == newPlaybackState)
+    {
+        return;
+    }
+    else
+    {
+        writePlaybackState(endpoint, newPlaybackState);
+    }
+}
+
+static void sendResponse(chip::app::Command * command, const char * responseName, chip::CommandId commandId,
+                         EmberAfMediaPlaybackStatus mediaPlaybackStatus)
+{
+    static_assert(std::is_same<std::underlying_type_t<EmberAfMediaPlaybackStatus>, uint8_t>::value, "Wrong enum size");
```

Comment:
Why is this assert needed?  We are no longer assuming things about the size of this enum, right?

Reply:
Removed.